examples/virus_on_network/app.py

```diff
@@ -0,0 +1,134 @@
+import math
+
+import solara
+from matplotlib.figure import Figure
+from matplotlib.ticker import MaxNLocator
+from mesa.experimental import JupyterViz, make_text
+from virus_on_network.model import State, VirusOnNetwork, number_infected
+
+
+def agent_portrayal(graph):
+    def get_agent(node):
+        return graph.nodes[node]["agent"][0]
+
+    width = []  # edge width
+    edge_color = []
+    for u, v in graph.edges():
```

Comment:
u, v, w and ec are quite not-descriptive. Maybe call it edge_width, edge_color here and above edge_colors*s* and edge_width*s* ?

Reply:
`edge_color` in NetworkX is a list of edge colors. As much as this is confusing, I have to stay consistent with NetworkX. Ditto with the plural for `edge_width`.

`u`, `v` should be fine for undirected graph, just like `i`, `j` for 2D grid.

Reply:
See `edge_color` in https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx.html, which may be a scalar or vector.